### ISSUE TITLE
[Feature]: Vue Support

### ISSUE THREAD

**Issue by atilkan** [remix-run/remix#442]
### What is the new or updated feature that you are suggesting?

Support Vue.js

### Why should this feature be included?

The same reason for React, Preact.

**Comment by egoist** [comment]
This is definitely out of the scope of this project 🥶

**Comment by sergiodxa** [comment]
> This is definitely out of the scope of this project 🥶

I don’t think so, Remix uses React but it’s not ready tired too it, if you see there’s a package with the related React code, so there could be another with Vue, and the same way Remix let you choose between Node or CF Workers, you could choose between React or Vue

**Comment by patrickcurl** [comment]
I agree with what @sergiodxa said... if you think about it...a framework that's a bit like next or remix that is frontend agnostic with simple hooks to basically interop between react, vue, svelte, alpine, etc would be a real game changer...

it'd be like if next and nuxt were one and the same with the same API just with options on vue SFCs or react components... instead of learning something new, there'd be a unified standard....

in a way I think inertiajs sort of does this at least for laravel and vue apps.. but I like the app structure with remix and it feels snappier than laravel+inertiajs+vue...

**Comment by Flexicon** [comment]
I'll chime in here as well with the fact that Remix could have the potential of being something like [Astro](https://astro.build/) but for actual apps, since Astro pretty much aims at building SSG websites.

**Comment by MichaelDeBoey** [comment]
I kind of agree with @sergiodxa on here, but I also see problems arising here:
Remix is built upon `react-router`, which is (obviously) a React library.

Supporting Vue would mean that we would need an API compatible `vue-router` kind of thing, which is not something the Remix team wants to maintain I think.
A third party could perfectly create a `remix-vue` though and the Remix team can point to that as sort of an endorsement, but I think that if we're being realistic that official (real) support will not happen as long as it's not officially maintained by the Remix team.

Nothing forbids the team to hire a Vue expert though 🤷‍♂️

**Comment by remix-vue** [comment]
I just started [remix-vue](https://github.com/remix-vue/remix-vue)

**Comment by patrickcurl** [comment]
> I kind of agree with @sergiodxa on here, but I also see problems arising here:
> Remix is built upon react-router, which is (obviously) a React library.
It'd be probably a diff library altogether, but maybe swapping react-router for inertiajs which supports react/vue but maybe lacks things needed for remix to work, IDK.... 

Honestly, a framework agnostic router would be tight and might be a good project unto itself, for anyone w/ the time/ability to focus on that...

**Comment by Brendan-csel** [comment]
I'm just going to tag [SolidJs](https://www.solidjs.com/) onto this issue.  It already has a [router](https://github.com/solidjs/solid-app-router) with API very similar to react-router 6.

**Comment by jacob-ebey** [comment]
> I kind of agree with @sergiodxa on here, but I also see problems arising here: Remix is built upon `react-router`, which is (obviously) a React library.
> 
> Supporting Vue would mean that we would need an API compatible `vue-router` kind of thing, which is not something the Remix team wants to maintain I think. A third party could perfectly create a `remix-vue` though and the Remix team can point to that as sort of an endorsement, but I think that if we're being realistic that official (real) support will not happen as long as it's not officially maintained by the Remix team.
> 
> Nothing forbids the team to hire a Vue expert though 🤷‍♂️

`react-router` actually has no dependency on react and is just a matching algorithm when it comes down to it and is something that 100% could be integrated into a vue, svelte, solid, etc, router. Remix itself is framework agnostic as well consisting of a server runtime based on fetch, and a client transition manager that is based on state updates that can be wired into any frontend framework. I'm excited myself to explore frameworks other than react and believe we are uniquely positioned to achieve this with relative ease compared to others.

**Comment by MichaelDeBoey** [comment]
> `react-router` actually has no dependency on react and is just a matching algorithm when it comes down to it and is something that 100% could be integrated into a vue, svelte, solid, etc, router.

`react-router` does have a peerDependency on `react` though
https://github.com/remix-run/react-router/blob/48b7c60a2133a617d64533b54eb299dbd2d1be0e/packages/react-router/package.json#L16-L18

**Comment by nullpaper** [comment]
+1 on the SolidJS port.

> @remix-vue
> I just started [remix-vue](https://github.com/remix-vue/remix-vue)

The word you're looking for is "squatted" not "started".
